src/vulkan-renderer/application.cpp

```diff
@@ -486,12 +488,37 @@ VkResult Application::update_mouse_input() {
     return VK_SUCCESS;
 }
 
+VkResult Application::update_imgui_overlay() {
+    ImGuiIO &io = ImGui::GetIO();
+
+    // TODO: Does that work? We can't just pass time_passed since it's 0 in the beginning and imgui doesn't accept that.
+    io.DeltaTime = std::clamp(m_time_passed, 0.001f, 100.0f);
+
+    double current_cursor_x;
+    double current_cursor_y;
+
+    glfwGetCursorPos(m_window->get(), &current_cursor_x, &current_cursor_y);
+
+    // TODO() move to update() method: mouse buttons left+right and cursor position
+    io.MousePos = ImVec2(static_cast<float>(current_cursor_x), static_cast<float>(current_cursor_y));
+    io.MouseDown[0] = (glfwGetMouseButton(m_window->get(), GLFW_MOUSE_BUTTON_LEFT) == GLFW_PRESS);
+    io.MouseDown[1] = (glfwGetMouseButton(m_window->get(), GLFW_MOUSE_BUTTON_RIGHT) == GLFW_PRESS);
+
+    if (m_imgui_overlay->update() || m_imgui_overlay->updated) {
+        m_imgui_overlay->prepare_drawing();
+        m_imgui_overlay->updated = false;
+    }
+
+    return VK_SUCCESS;
```

Comment:
Making this function return `VK_SUCCESS` doesn't really add anything.

Reply:
Right. This is part of the bigger picture of https://github.com/inexorgame/vulkan-renderer/issues/122 which tells us to use exceptions instead of return codes for error handling.

Reply:
Done.